spec.md

```diff
@@ -64,42 +64,72 @@ We also use the following operators:
 The primary result of this specification is to define a family of
 functions:
 
-$\operatorname{SPLIT}_C \in V_8 \rightarrow V_v$
+$\operatorname{SPLIT}_{T, C} \in V_8 \rightarrow V_v$
 
-...which is parameterized by a configuration $C$, consisting of:
+...which is parameterized by an integer $T \in U_{32}$, called the
+"threshold", and a configuration $C$, consisting of:
 
 - $S_{\text{min}} \in U_{32}$, the minimum split size
 - $S_{\text{max}} \in U_{32}$, the maximum split size
 - $H \in V_8 \rightarrow U_{32}$, the hash function
 - $W \in U_{32}$, the window size
-- $T \in U_{32}$, the threshold
 
 The configuration must satisfy $S_{\text{max}} \ge S_{\text{min}} \ge W$.
 
 ## Definitions
 
-The "split index" $I(X)$, if it exists, is the smallest
+The "split index" $I_{T, C}(X)$, if it exists, is the smallest
 integer $i$ satisfying each of:
 
 - $i < |X|$
 - $S_{\text{max}} \ge i \ge S_{\text{min}}$
 - $H(\langle X_{i-W+1}, \dots, X_i \rangle) \mod 2^T = 0$
 
-We define $\operatorname{SPLIT}_C(X)$ recursively, as follows:
+We define $\operatorname{SPLIT}_{T, C}(X)$ recursively, as follows:
 
-- If $|X| = 0$, $\operatorname{SPLIT}_C(X) = \langle \rangle$
-- Otherwise, if $I(X)$ exists,
-  $\operatorname{SPLIT}_C(X) = \langle Y \rangle \mathbin{\|} \operatorname{SPLIT}_C(Z)$
-  where
-  - $i = I(X)$
+- If $|X| = 0$, $\operatorname{SPLIT}_{T, C}(X) = \langle \rangle$
+- Otherwise, if $I_{T, C}(X)$ exists, $\operatorname{SPLIT}_{T, C}(X) =
+  \langle Y \rangle \mathbin{\|} \operatorname{SPLIT}_{T, C}(Z)$ where
+  - $i = I_{T, C}(X)$
   - $N = |X| - 1$
   - $Y = \langle X_0, \dots, X_i \rangle$
   - $Z = \langle X_{i+1}, \dots, X_N \rangle$
-- Otherwise, $\operatorname{SPLIT}_C(X) = \langle X \rangle$.
+- Otherwise, $\operatorname{SPLIT}_{T, C}(X) = \langle X \rangle$.
 
 # Tree Construction
 
-TODO
+For our purposes, a tree consists of a node (the "root") with an
+associated value (its "label") along with zero or more children, each
+of which is itself a tree. The children of each node are ordered. A
+"leaf" is a node with zero children; for convenience, a tree whose root
+is a leaf node will also be referred to as a "leaf".
+
+Let $C$ be a configuration, and let $T$ and $S$ be positive integers. We
+call $T$ the "threshold", as before, and $S$ the "step". Now for $X \in
+V_8$ we define $\operatorname{TREE}_{T, S, C} (X)$ recursively as
+follows:
+
+- If $|X| = 0$, $\operatorname{TREE}_{T, S, C} (X)$ is a leaf labelled
+  by $X$
+- Otherwise, if $I_{T, C}(X)$ exists, $\operatorname{TREE}_{T, S, C}
+  (X)$ is a node labelled by $X$ whose children in order are leaves
+  corresponding to, and labelled by, the entries of
+  $\operatorname{SPLIT}_{T, C} (X)$
+- Otherwise, $\operatorname{TREE}_{T, S, C}$ is a node labelled by $X$
+  whose children in order are the subtrees $\operatorname{TREE}_{T, S,
+  C} (Y)$ for each entry $Y$ in $\operatorname{SPLIT}_{T + nS, C} (X)$,
+  where $n$ is the largest integer such that $I_{T + nS, C} (X)$
+  exists.
```

Comment:
If I'm reading this correctly, I don't think this construction quite works as intended: S_max is going to bound the maximum size of a chunk, so you can't just apply split with a greater T but the same C and expect it to group the smaller chunks. 

I think this is the right way to approach this conceptually, but that interaction needs fixing.

Reply:
You're quite right, though it took me a little while to see it. I have a fix in mind, will update once it's written up.